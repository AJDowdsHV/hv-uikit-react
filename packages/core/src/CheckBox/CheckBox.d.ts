import { StandardProps } from "@material-ui/core";
import React from "react";

import { HvBaseCheckBoxProps } from "../BaseCheckBox";
import { HvFormStatus } from "../Forms/FormElement";
import { HvLabelProps } from "../Forms/Label";

export type HvCheckBoxClassKey =
  | "root"
  | "container"
  | "disabled"
  | "checkBox"
  | "label"
  | "focusVisible"
  | "labelTypography";

export interface HvCheckBoxProps extends StandardProps<HvBaseCheckBoxProps, HvCheckBoxClassKey> {
  /**
   * The label of the form element.
   *
   * The form element must be labeled for accessibility reasons.
   * If not provided, an aria-label or aria-labelledby must be inputted via inputProps.
   */
  label?: React.ReactNode;
  /**
   * Properties passed on to the label element.
   */
  labelProps?: HvLabelProps;
  /**
   * The status of the form element.
   *
   * Valid is correct, invalid is incorrect and standBy means no validations have run.
   *
   * When uncontrolled and unspecified it will default to "standBy" and change to either "valid"
   * or "invalid" after any change to the state.
   */
  status?: HvFormStatus;
  /**
   * The error message to show when `status` is "invalid". Defaults to "Required".
   */
  statusMessage?: React.ReactNode;
}

export default function HvCheckBox(props: HvCheckBoxProps): JSX.Element | null;
