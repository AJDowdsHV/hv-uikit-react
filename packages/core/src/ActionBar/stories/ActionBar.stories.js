import React from "react";
import { makeStyles } from "@material-ui/core";
import { HvActionContainer, HvButton, HvDropDownMenu } from "../..";

export default {
  title: "Patterns/Action Bar/Action Container",
  parameters: {
    componentSubtitle: null,
    usage: "import { HvActionContainer } from '@hv/uikit-react-core/dist'"
  },
  component: HvActionContainer
};

export const Main = () => {
  const useStyles = makeStyles(theme => ({
    root: {
      margin: "auto",
      backgroundColor: theme.hv.palette.atmosphere.atmo1,
      width: "1000px"
    },
    flexBox: {
      display: "flex"
    },
    separator: {
      marginRight: theme.hv.spacing.sm
    }
  }));

  const classes = useStyles();

  return (
    <div className={classes.root}>
      <HvActionContainer id="ActionContainer">
        <div className={classes.flexBox}>
          <div className={classes.separator}>
            <HvButton category="ghost" onClick={() => console.log("Save action")}>
              Save
            </HvButton>
          </div>
          <div>
            <HvButton category="ghost" onClick={() => console.log("Cancel action")}>
              Cancel
            </HvButton>
          </div>
        </div>
        <div>
          <HvButton category="ghost" onClick={() => console.log("Help action")}>
            Help
          </HvButton>
        </div>
      </HvActionContainer>
    </div>
  );
};

export const DualAction = () => {
  const useStyles = makeStyles(theme => ({
    root: {
      margin: "auto",
      backgroundColor: theme.hv.palette.atmosphere.atmo1,
      width: "1000px"
    },
    flexBox: {
      display: "flex"
    },
    separator: {
      marginRight: theme.hv.spacing.sm
    }
  }));

  const classes = useStyles();

  return (
    <div className={classes.root}>
      <HvActionContainer id="ActionContainer">
        <div className={classes.flexBox}>
          <div className={classes.separator}>
            <HvButton category="ghost" onClick={() => console.log("Save action")}>
              Save
            </HvButton>
          </div>
          <div>
            <HvButton category="ghost" onClick={() => console.log("Cancel action")}>
              Cancel
            </HvButton>
          </div>
        </div>
      </HvActionContainer>
    </div>
  );
};

DualAction.story = {
  parameters: {
    docs: {
      storyDescription: "Showcasing the action bar pattern with only two actions."
    }
  }
};

export const VariedActionBar = () => {
  const useStyles = makeStyles(theme => ({
    root: {
      margin: "auto",
      backgroundColor: theme.hv.palette.atmosphere.atmo1,
      width: "1000px"
    },
    flexBox: {
      display: "flex"
    },
    separator: {
      marginRight: theme.hv.spacing.sm
    }
  }));

  const classes = useStyles();

  return (
    <div className={classes.root}>
      <HvActionContainer id="ActionContainer">
        <div className={classes.flexBox}>
          <div className={classes.separator}>
            <HvButton category="ghost" onClick={() => console.log("Save action")}>
              Save
            </HvButton>
          </div>
          <div>
            <HvDropDownMenu
              onClick={(e, item) => console.log(item.label)}
              dataList={[{ label: "Delete" }, { label: "Update" }]}
            />
          </div>
        </div>
        <div>
          <HvButton category="ghost" onClick={() => console.log("Help action")}>
            Help
          </HvButton>
        </div>
      </HvActionContainer>
    </div>
  );
};

VariedActionBar.story = {
  parameters: {
    docs: {
      storyDescription: "Showcasing the action bar pattern with only two actions."
    }
  }
};
