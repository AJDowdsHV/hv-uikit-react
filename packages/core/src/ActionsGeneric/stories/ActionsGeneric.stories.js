/* eslint-disable react/prop-types */
import React from "react";
import { Add, Delete, Lock, Preview } from "@hv/uikit-react-icons";
import HvActionsGeneric from "..";

export default {
  title: "Components/Actions Generic",
  parameters: {
    componentSubtitle: null,
    usage: "import { HvActionsGeneric } from '@hv/uikit-react-core/dist'",
    dsVersion: "3.2.0",
  },
  component: HvActionsGeneric,
  decorators: [
    (Story) => (
      <div
        style={{
          display: "flex",
          justifyContent: "center",
        }}
      >
        <Story />
      </div>
    ),
  ],
};

export const Main = () => (
  <HvActionsGeneric
    actions={[
      { id: "add", label: "Add", iconCallback: () => <Add /> },
      { id: "delete", label: "Delete", iconCallback: () => <Delete /> },
      { id: "lock", label: "Lock", iconCallback: () => <Lock /> },
      { id: "put", label: "Preview", iconCallback: () => <Preview /> },
    ]}
    maxVisibleActions={2}
  />
);
