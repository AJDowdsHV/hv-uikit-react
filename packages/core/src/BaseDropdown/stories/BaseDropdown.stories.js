import React, { useState } from "react";
import { MoreOptionsVertical } from "@hv/uikit-react-icons";
import { useTheme } from "@material-ui/core";
import HvBaseDropdown from "..";
import HvInput from "../../Input";
import HvButton from "../../Button";
import HvTypography from "../../Typography";
import { HvFormElement, HvInfoMessage, HvLabel } from "../../Forms";

export default {
  title: "Components/BaseDropdown",
  parameters: {
    componentSubtitle: null,
    usage: "import { HvBaseDropdown } from '@hv/uikit-react-core/dist'"
  },
  component: HvBaseDropdown
};

export const Main = () => <HvBaseDropdown placeholder="Placeholder..." />;

Main.story = {
  parameters: {
    v3: true,
    pa11y: {
      ignore: ["region"]
    }
  }
};

export const WithContent = () => {
  const theme = useTheme();
  return (
    <HvBaseDropdown
      placeholder={<HvTypography id="labelTypography">Placeholder</HvTypography>}
      aria-labelledby="labelTypography"
    >
      <div style={{ width: "300px", height: "300px", background: theme.palette.atmo2 }}>
        <HvInput id="input-dropdown" />
      </div>
    </HvBaseDropdown>
  );
};

WithContent.story = {
  parameters: {
    v3: true
  }
};

export const WithForms = () => (
  <HvFormElement id="withForms" disabled={false}>
    <HvLabel style={{ paddingBottom: "8px", display: "block" }} label="Label" />
    <HvBaseDropdown placeholder="Placeholder..." />
    <HvInfoMessage disableGutter style={{ paddingTop: "8px", display: "block" }}>
      The description.
    </HvInfoMessage>
  </HvFormElement>
);

WithForms.story = {
  parameters: {
    v3: true
  }
};

export const Controlled = () => {
  const [open, setOpen] = useState(false);

  return (
    <>
      <HvButton onClick={() => setOpen(!open)}>change</HvButton>
      <p />
      <HvBaseDropdown
        placeholder="PlaceHolder..."
        expanded={open}
        onToggle={(e, s) => setOpen(s)}
      />
    </>
  );
};

Controlled.story = {
  parameters: {
    v3: true
  }
};

export const Custom = () => {
  const [open, setOpen] = useState(false);

  const theme = useTheme();

  const setterOpen = () => setOpen(!open);

  return (
    <HvBaseDropdown
      expanded={open}
      onClickOutside={setterOpen}
      component={
        <HvButton
          overrideIconColors={false}
          onClick={setterOpen}
          category="icon"
          aria-label="Dropdown menu"
          style={{ backgroundColor: open ? theme.palette.atmo1 : undefined }}
        >
          <MoreOptionsVertical />
        </HvButton>
      }
    >
      <div style={{ width: "500px", height: "500px", background: theme.palette.atmo2 }} />
    </HvBaseDropdown>
  );
};

Custom.story = {
  parameters: {
    v3: true
  }
};
