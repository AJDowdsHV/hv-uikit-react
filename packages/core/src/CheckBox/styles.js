import { outlineStyles } from "../Focus/styles";

const styles = (theme) => ({
  root: {
    display: "inline-block",
  },

  container: {
    cursor: "pointer",
    display: "flex",

    "&:hover": {
      backgroundColor: theme.hv.palette.atmosphere.atmo3,
    },
  },
  disabled: {
    cursor: "not-allowed",
  },

  checkbox: {},

  label: {
    // ensure more specificity than .HvTypography-highlightText
    "$root &": {
      overflow: "hidden",
      textOverflow: "ellipsis",
      verticalAlign: "middle",
      paddingRight: theme.hv.spacing.xs,
      whiteSpace: "nowrap",
      ...theme.hv.typography.normalText,
      cursor: "pointer",
      height: "32px",
      lineHeight: "32px",
      width: "100%",
    },
    "$disabled &": {
      // not spreading theme.hv.typography.placeholderText, it overrides too many things
      color: theme.hv.palette.atmosphere.atmo5,
      cursor: "not-allowed",
    },
  },
  focusVisible: {
    ...outlineStyles,
    backgroundColor: theme.hv.palette.atmosphere.atmo3,

    // ensure more specificity than .HvBaseCheckbox-root.HvBaseCheckbox-focusVisible svg
    "& $checkbox div>svg": {
      outline: "none",
      boxShadow: "none",
    },
  },
});

export default styles;
