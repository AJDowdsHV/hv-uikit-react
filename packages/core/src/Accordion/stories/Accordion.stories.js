import React, { useState, useMemo } from "react";
import { makeStyles } from "@mui/styles";
import {
  HvAccordion,
  HvListContainer,
  HvListItem,
  HvInput,
  HvButton,
  HvTable,
  HvTableHead,
  HvTableHeader,
  HvTableRow,
  HvTableBody,
  HvTableCell,
  useHvData,
  useHvTableSticky,
} from "../..";

export default {
  title: "Display/Accordion",
  parameters: {
    componentSubtitle: null,
    usage: "import { HvAccordion } from '@hitachivantara/uikit-react-core'",
  },

  component: HvAccordion,
};

export const Main = () => {
  const useStyles = makeStyles(() => ({
    accordionSpacing: {
      "& > li": {
        paddingLeft: 32,
      },
    },
    accordionContainer: {
      maxWidth: 300,
    },
  }));
  const classes = useStyles();
  return (
    <div className={classes.accordionContainer}>
      <HvAccordion id="item1" label="Analytics" headingLevel={3}>
        <HvListContainer className={classes.accordionSpacing} interactive condensed>
          <HvListItem>Views</HvListItem>
          <HvListItem>Parameters</HvListItem>
        </HvListContainer>
      </HvAccordion>
      <HvAccordion id="item2" label="System" headingLevel={3}>
        <HvListContainer className={classes.accordionSpacing} interactive condensed>
          <HvListItem>Settings</HvListItem>
          <HvListItem>Network</HvListItem>
        </HvListContainer>
      </HvAccordion>
      <HvAccordion id="item3" label="Data" headingLevel={3}>
        <HvListContainer className={classes.accordionSpacing} interactive condensed>
          <HvListItem>Storage</HvListItem>
          <HvListItem>Memory</HvListItem>
        </HvListContainer>
      </HvAccordion>
    </div>
  );
};

export const Disabled = () => {
  const useStyles = makeStyles(() => ({
    accordionSpacing: {
      "& > li": {
        paddingLeft: 32,
      },
    },
    accordionContainer: {
      maxWidth: 300,
    },
  }));
  const classes = useStyles();
  return (
    <div className={classes.accordionContainer}>
      <HvAccordion id="disabled-item1" label="Analytics" headingLevel={3} disabled>
        <HvListContainer className={classes.accordionSpacing} interactive condensed>
          <HvListItem>Views</HvListItem>
          <HvListItem>Parameters</HvListItem>
        </HvListContainer>
      </HvAccordion>
      <HvAccordion id="disabled-item2" label="System" headingLevel={3}>
        <HvListContainer className={classes.accordionSpacing} interactive condensed>
          <HvListItem>Settings</HvListItem>
          <HvListItem>Network</HvListItem>
        </HvListContainer>
      </HvAccordion>
      <HvAccordion id="disabled-item3" label="Data" headingLevel={3} disabled>
        <HvListContainer className={classes.accordionSpacing} interactive condensed>
          <HvListItem>Storage</HvListItem>
          <HvListItem>Memory</HvListItem>
        </HvListContainer>
      </HvAccordion>
    </div>
  );
};

Disabled.parameters = {
  docs: {
    description: { story: "A common accordion with disable items. " },
  },
};

export const Controlled = () => {
  const useStyles = makeStyles((theme) => ({
    accordionSpacing: {
      "& > li": {
        paddingLeft: 32,
      },
    },
    accordionContainer: {
      maxWidth: 300,
    },
    buttonContainer: {
      "& button": {
        marginRight: theme.hvSpacing("xs"),
        marginBottom: theme.hvSpacing("md"),
      },
    },
  }));
  const classes = useStyles();
  const [expandedState, setExpandedState] = useState({ analytics: true, system: true, data: true });
  const handleToggle = (key) => {
    const newValue = { ...expandedState };
    newValue[key] = !newValue[key];
    setExpandedState(newValue);
  };
  const handleAll = (option) => {
    setExpandedState({ analytics: option, system: option, data: option });
  };
  return (
    <>
      <div className={classes.buttonContainer}>
        <HvButton category="secondary" onClick={() => handleToggle("analytics")}>
          Toggle Analytics
        </HvButton>
        <HvButton category="secondary" onClick={() => handleToggle("system")}>
          Toggle System
        </HvButton>
        <HvButton category="secondary" onClick={() => handleToggle("data")}>
          Toggle Data
        </HvButton>
        <HvButton category="secondary" onClick={() => handleAll(false)}>
          Close all
        </HvButton>
        <HvButton category="secondary" onClick={() => handleAll(true)}>
          Expand all
        </HvButton>
      </div>
      <div className={classes.accordionContainer}>
        <HvAccordion
          id="controlled-item1"
          label="Analytics"
          onChange={() => handleToggle("analytics")}
          expanded={expandedState.analytics}
        >
          <HvListContainer className={classes.accordionSpacing} interactive condensed>
            <HvListItem>Views</HvListItem>
            <HvListItem>Parameters</HvListItem>
          </HvListContainer>
        </HvAccordion>
        <HvAccordion
          id="controlled-item2"
          label="System"
          onChange={() => handleToggle("system")}
          expanded={expandedState.system}
        >
          <HvListContainer className={classes.accordionSpacing} interactive condensed>
            <HvListItem>Settings</HvListItem>
            <HvListItem>Network</HvListItem>
          </HvListContainer>
        </HvAccordion>
        <HvAccordion
          id="controlled-item3"
          label="Data"
          onChange={() => handleToggle("data")}
          expanded={expandedState.data}
        >
          <HvListContainer className={classes.accordionSpacing} interactive condensed>
            <HvListItem>Storage</HvListItem>
            <HvListItem>Memory</HvListItem>
          </HvListContainer>
        </HvAccordion>
      </div>
    </>
  );
};

Controlled.parameters = {
  docs: {
    description: { story: "An accordion controlled externally forcing expansion." },
  },
};

export const Form = () => {
  const useStyles = makeStyles(() => ({
    accordionContainer: {
      maxWidth: 300,
    },
    formContainer: {
      padding: "0 32px",
      "& > div": {
        paddingTop: 17,
      },
    },
  }));
  const classes = useStyles();
  return (
    <div className={classes.accordionContainer}>
      <HvAccordion id="personal" label="Personal Information" headingLevel={3}>
        <div className={classes.formContainer}>
          <HvInput id="input-name" label="Name" placeholder="Insert first name" required />
          <HvInput id="input-email" label="Email" placeholder="Insert your email" required />
          <HvInput id="input-phone" label="Phone" placeholder="Insert your phone number" />
          <HvInput id="input-extension" label="Extension" placeholder="Insert phone extension" />
          <HvInput id="input-country" label="Country" placeholder="Insert country name" />
          <HvInput id="input-province" label="City/Province" placeholder="Insert province name" />
        </div>
      </HvAccordion>
      <HvAccordion id="billing" label="Billing Address" headingLevel={3}>
        <div className={classes.formContainer}>
          <HvInput id="input-address" label="Address 1" placeholder="Insert first name" />
          <HvInput id="input-address2" label="Address 2" placeholder="Insert address" />
          <HvInput id="input-city" label="City" placeholder="Insert city name" />
          <HvInput id="input-state" label="State" placeholder="Insert state" />
          <HvInput id="input-code" label="Zip Code" placeholder="Insert code" />
        </div>
      </HvAccordion>
      <HvAccordion id="shipping" label="Shipping Address" headingLevel={3}>
        <div className={classes.formContainer}>
          <HvInput id="input-address-bill" label="Address 1" placeholder="Insert first name" />
          <HvInput id="input-address2-bill" label="Address 2" placeholder="Insert address" />
          <HvInput id="input-city-bill" label="City" placeholder="Insert city name" />
          <HvInput id="input-state-bill" label="State" placeholder="Insert state" />
          <HvInput id="input-code-bill" label="Zip Code" placeholder="Insert code" />
        </div>
      </HvAccordion>
    </div>
  );
};

Form.parameters = {
  docs: {
    description: { story: "An accordion composition with forms." },
  },
};

export const Typography = () => {
  const useStyles = makeStyles(() => ({
    accordionContainer: {
      width: "100%",
    },
  }));
  const classes = useStyles();

  const getData = () => [
    {
      id: 1,
      title: "A new hope",
      director: "George Lucas",
      producer: "Gary Kurzt",
      release: "1977-05-25",
    },
    {
      id: 2,
      title: "Empires strike back",
      director: "Irvin Kershner",
      producer: "Gary Kurzt",
      release: "1980-05-17",
    },
    {
      id: 3,
      title: "Return of the jedi",
      director: "Richard Marquand",
      producer: "Howard Kazanjian",
      release: "1983-05-25",
    },
  ];

  const getColumns = () => [
    {
      Header: "Title",
      accessor: "title",
      sticky: "left",
      width: 150,
      style: { textTransform: "capitalize" },
    },
    {
      Header: "Director",
      accessor: "director",
    },
    {
      Header: "Producer",
      accessor: "producer",
    },
    {
      Header: "Release Date",
      accessor: "release",
    },
  ];

  const data = useMemo(() => getData(), []);
  const columns = useMemo(() => getColumns(), []);

  const { getTableProps, getTableHeadProps, getTableBodyProps, prepareRow, headerGroups, rows } =
    useHvData(
      {
        columns,
        data,
        stickyHeader: true,
      },
      useHvTableSticky
    );

  return (
    <div className={classes.accordionContainer}>
      <HvAccordion id="film" label="Films" headingLevel={2}>
        <HvTable id="accordion-table" {...getTableProps()}>
          <HvTableHead {...getTableHeadProps()}>
            {headerGroups.map((headerGroup) => (
              <HvTableRow {...headerGroup.getHeaderGroupProps()}>
                {headerGroup.headers.map((col) => (
                  <HvTableHeader {...col.getHeaderProps()}>{col.render("Header")}</HvTableHeader>
                ))}
              </HvTableRow>
            ))}
          </HvTableHead>
          <HvTableBody {...getTableBodyProps()}>
            {rows.map((row) => {
              prepareRow(row);

              return (
                <HvTableRow {...row.getRowProps()}>
                  {row.cells.map((cell) => (
                    <HvTableCell {...cell.getCellProps()}>{cell.render("Cell")}</HvTableCell>
                  ))}
                </HvTableRow>
              );
            })}
          </HvTableBody>
        </HvTable>
      </HvAccordion>
    </div>
  );
};

Typography.parameters = {
  docs: {
    description: { story: "An accordion with a different button typography." },
  },
};
