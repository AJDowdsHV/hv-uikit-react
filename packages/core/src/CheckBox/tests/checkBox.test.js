/* eslint-env jest */
import React from "react";
import { mount } from "enzyme";

import { HvCheckBox, HvProvider } from "../..";

import { Main } from "../stories/CheckBox.stories";

describe("CheckBox", () => {
  let wrapper;

  beforeEach(async () => {
    wrapper = mount(
      <HvProvider>
        <Main />
      </HvProvider>
    );
  });

  it("should render correctly", () => {
    expect(wrapper.find(HvCheckBox)).toMatchSnapshot();
  });

  it("should render the Checkbox components", () => {
    const buttonComponent = wrapper.find(HvCheckBox);
    expect(buttonComponent.length).toBe(3);
  });
});
