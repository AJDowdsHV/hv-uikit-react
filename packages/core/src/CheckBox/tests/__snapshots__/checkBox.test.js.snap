// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HvCheckBox sample snapshot testing ErrorMessage 1`] = `
<div>
  <div
    class="HvCheckBox-root-43 HvFormElement-root-51"
  >
    <div
      class="HvCheckBox-container-44 HvCheckBox-invalidContainer-45"
    >
      <span
        class="MuiCheckbox-root MuiCheckbox-colorDefault MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorDefault PrivateSwitchBase-root HvCheckBox-checkbox-47 HvCheckBox-invalidCheckbox-48 HvBaseCheckbox-root-52 css-balhlm-MuiButtonBase-root-MuiCheckbox-root"
      >
        <input
          aria-errormessage="hvcheckbox7-error"
          aria-invalid="true"
          class="PrivateSwitchBase-input css-1m9pwf3"
          data-indeterminate="false"
          id="hvcheckbox7-input"
          type="checkbox"
          value="on"
        />
        <div
          class="HvIconBase-root-55 HvIconBase-s-57"
          font-size="medium"
          name="Checkbox"
        >
          <svg
            focusable="false"
            height="16"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              class="color0"
              d="M0 0h16v16H0z"
              fill="#FBFCFC"
            />
            <path
              class="color1"
              d="M16 16H0V0h16zM1 15h14V1H1z"
              fill="#414141"
            />
          </svg>
        </div>
      </span>
      <label
        class="HvCheckBox-label-49 HvLabel-root-60 HvTypography-root-63 HvTypography-highlightText-75"
        for="hvcheckbox7-input"
        id="hvcheckbox7-label"
      >
        Checkbox 1
      </label>
    </div>
    <div
      class="HvWarningText-root-85 HvWarningText-show-90"
    >
      <div
        class="HvWarningText-defaultIcon-89 HvIconBase-root-55 HvIconBase-s-57"
        name="Fail"
      >
        <svg
          focusable="false"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            class="color0"
            d="M7.5 4.001h1v6h-1zm0 8h1v-1h-1zM16 8a8 8 0 10-8 8 8 8 0 008-8zm-1 0a7 7 0 11-7-7 7 7 0 017 7z"
            fill="#C62828"
          />
        </svg>
      </div>
      <p
        aria-atomic="true"
        aria-live="polite"
        aria-relevant="additions text"
        class="HvWarningText-warningText-88 HvWarningText-topGutter-87 HvTypography-root-63 HvTypography-normalText-76"
        id="hvcheckbox7-error"
      >
        No way for this to be valid!
      </p>
    </div>
  </div>
</div>
`;

exports[`HvCheckBox sample snapshot testing Main 1`] = `
<div>
  <div
    class="HvCheckBox-root-1 HvFormElement-root-9"
  >
    <div
      class="HvCheckBox-container-2"
    >
      <span
        class="MuiCheckbox-root MuiCheckbox-colorDefault MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorDefault PrivateSwitchBase-root HvCheckBox-checkbox-5 HvBaseCheckbox-root-10 css-balhlm-MuiButtonBase-root-MuiCheckbox-root"
      >
        <input
          class="PrivateSwitchBase-input css-1m9pwf3"
          data-indeterminate="false"
          id="hvcheckbox1-input"
          type="checkbox"
          value="on"
        />
        <div
          class="HvIconBase-root-13 HvIconBase-s-15"
          font-size="medium"
          name="Checkbox"
        >
          <svg
            focusable="false"
            height="16"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              class="color0"
              d="M0 0h16v16H0z"
              fill="#FBFCFC"
            />
            <path
              class="color1"
              d="M16 16H0V0h16zM1 15h14V1H1z"
              fill="#414141"
            />
          </svg>
        </div>
      </span>
      <label
        class="HvCheckBox-label-7 HvLabel-root-18 HvTypography-root-21 HvTypography-highlightText-33"
        for="hvcheckbox1-input"
        id="hvcheckbox1-label"
      >
        Checkbox 1
      </label>
    </div>
  </div>
  <div
    class="HvCheckBox-root-1 HvFormElement-root-9"
  >
    <div
      class="HvCheckBox-container-2"
    >
      <span
        class="MuiCheckbox-root MuiCheckbox-colorDefault MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorDefault PrivateSwitchBase-root Mui-checked HvCheckBox-checkbox-5 HvBaseCheckbox-root-10 css-balhlm-MuiButtonBase-root-MuiCheckbox-root"
      >
        <input
          checked=""
          class="PrivateSwitchBase-input css-1m9pwf3"
          data-indeterminate="false"
          id="hvcheckbox3-input"
          type="checkbox"
          value="on"
        />
        <div
          class="HvIconBase-root-13 HvIconBase-s-15"
          font-size="medium"
          name="CheckboxCheck"
        >
          <svg
            focusable="false"
            height="16"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              class="color0"
              d="M0 0h16v16H0z"
              fill="#414141"
            />
            <path
              class="color0"
              d="M16 16H0V0h16zM1 15h14V1H1z"
              fill="#414141"
            />
            <path
              class="color1"
              d="M6.049 12.463L2.293 8.707l1.414-1.414 2.244 2.244 6.3-7.2 1.506 1.317z"
              fill="#FBFCFC"
            />
          </svg>
        </div>
      </span>
      <label
        class="HvCheckBox-label-7 HvLabel-root-18 HvTypography-root-21 HvTypography-highlightText-33"
        for="hvcheckbox3-input"
        id="hvcheckbox3-label"
      >
        Checkbox 2
      </label>
    </div>
  </div>
  <div
    class="HvCheckBox-root-1 HvFormElement-root-9"
  >
    <div
      class="HvCheckBox-container-2"
    >
      <span
        class="MuiCheckbox-root MuiCheckbox-indeterminate MuiCheckbox-colorDefault MuiButtonBase-root MuiCheckbox-root MuiCheckbox-indeterminate MuiCheckbox-colorDefault PrivateSwitchBase-root HvCheckBox-checkbox-5 HvBaseCheckbox-root-10 css-balhlm-MuiButtonBase-root-MuiCheckbox-root"
      >
        <input
          class="PrivateSwitchBase-input css-1m9pwf3"
          data-indeterminate="true"
          id="hvcheckbox5-input"
          type="checkbox"
          value="on"
        />
        <div
          class="HvIconBase-root-13 HvIconBase-s-15"
          font-size="medium"
          name="CheckboxPartial"
        >
          <svg
            focusable="false"
            height="16"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              class="color0"
              d="M0 0h16v16H0z"
              fill="#FBFCFC"
            />
            <path
              class="color1"
              d="M16 16H0V0h16zM1 15h14V1H1zm3-6h8V7H4z"
              fill="#414141"
            />
          </svg>
        </div>
      </span>
      <label
        class="HvCheckBox-label-7 HvLabel-root-18 HvTypography-root-21 HvTypography-highlightText-33"
        for="hvcheckbox5-input"
        id="hvcheckbox5-label"
      >
        Checkbox 3
      </label>
    </div>
  </div>
</div>
`;
